velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -47,10 +47,14 @@ struct Date {
   int32_t dayOfYear = 1;
   bool dayOfYearFormat = false;
 
+  int32_t weekOfMonth = 1;
+  bool weekOfMonthDateFormat = false;
+
   bool centuryFormat = false;
 
   bool isYearOfEra = false; // Year of era cannot be zero or negative.
   bool hasYear = false; // Whether year was explicitly specified.
+  bool hasDayOfWeek = false; // Whether dayOfWeek was explicitly specified.
```

Comment:
unrelated to this PR, but maybe we should use `std::optional` to prevent these flags controlling what has been explicitly set and what not.

Reply:
Thanks for your suggestion! Using std::optional would be a more elegant solution to avoid using boolean flags to control whether a value has been explicitly set. I'd like to work on it when I have some time. Should I open an issue for this first?

Reply:
Thanks for your suggestion! Using std::optional would be a more elegant solution to avoid using boolean flags to control whether a value has been explicitly set. I'd like to work on it when I have some time. Should I open an issue for this first?